File: UCSample/UCSample/SettingCell.m
//
//  SettingCell.m
//  UCSample
//
//  Created by Joe Lepple on 9/25/13.
//  Copyright (c) 2013 PortSIP Solutions, Inc. All rights reserved.
//

#import "SettingCell.h"

@implementation SettingItem

@end

@implementation SettingCell

- (id)initWithStyle:(UITableViewCellStyle)style
    reuseIdentifier:(NSString *)reuseIdentifier {
  self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
  if (self) {
    // Initialization code
  }
  return self;
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
  [super setSelected:selected animated:animated];

  // Configure the view for the selected state
}

- (void)SetItem:(SettingItem *)item {
  settingItem = item;

  _switchOperation.on = item.enable;
  _switchOperation.tag = item.index;
  _nameLabel.text = item.name;
}

- (IBAction)onSwitchChange:(UISwitch *)sender {
  if (settingItem != nil) {
    settingItem.enable = [sender isOn];
  }
}
@end

File: UCSample_swift/UCSample/SettingCell.swift
//
//  SettingCell.m
//  UCSample
//
//  Created by Joe Lepple on 9/25/13.
//  Copyright (c) 2013 PortSIP Solutions, Inc. All rights reserved.
//
import UIKit

class SettingItem {
    var index: Int
    var name: String
    var enable: Bool
    var codeType: Int32

    init() {
        index = 0
        name = ""
        enable = false
        codeType = 0
    }
}

class SettingCell: UITableViewCell {
    var settingItem: SettingItem!
    @IBOutlet var nameLabel: UILabel!
    @IBOutlet var switchOperation: UISwitch!

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }

    func SetItem(_ item: SettingItem) {
        settingItem = item

        switchOperation.isOn = item.enable
        switchOperation.tag = item.index
        nameLabel.text = item.name
    }

    @IBAction func onSwitchChange(_ sender: AnyObject) {
        if settingItem != nil {
            settingItem.enable = (sender as! UISwitch).isOn
        }
    }
}
